src/main/java/meowmeow/Storage.java

```diff
@@ -0,0 +1,142 @@
+        //Create save file
+        try {
+            saveFile = new File(filePath);
+            File parent = saveFile.getParentFile();
+
+            if (!parent.exists()) {
+                parent.mkdirs();
+            }
+
+            saveFile.createNewFile();
+
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void save(ArrayList taskList) {
+        this.taskList = taskList;
+        PrintWriter printWriter = null;
+        try {
+            printWriter = new PrintWriter(saveFile);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        PrintWriter finalPrintWriter = printWriter;
+
+        taskList.forEach((t) -> {
+            Task k = (Task) t;
+            finalPrintWriter.println(k.getSaveData());
+        });
+        printWriter.close();
+    }
+
+    public ArrayList<Task> parseSaveFile(File txt) {
+        try {
+            txt.getAbsolutePath();
+            txt.createNewFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        Scanner sc = null;
+        try {
+            sc = new Scanner(txt);
+            while (sc.hasNextLine()) {
+                String text = sc.nextLine();
+
+                String[] split = text.split(" \\| ");
+                assert split.length > 0 : "Save file is empty";
+                String firstChar = split[0];
+                assert firstChar == "T" || firstChar == "D" || firstChar == "E" : "Save file is corrupted";
+
+                switch (firstChar) {
+                case "T":
+                    String taskName = split[2];
+
+                    Task todo = new ToDo(taskName);
+                    taskList.add(todo);
+
+                    boolean isDone = Boolean.parseBoolean(split[1]);
+                    if (isDone) {
+                        todo.markAsDone();
+                    }
+                    break;
+
+                case "D":
+                    taskName = split[2];
+
+                    LocalDateTime date = LocalDateTime.parse(split[3]);
+
+                    Task deadline = new Deadline(taskName, date);
+                    taskList.add(deadline);
+
+                    isDone = Boolean.parseBoolean(split[1]);
+                    if (isDone) {
+                        deadline.markAsDone();
+                    }
+                    break;
+
+                case "E":
+                    taskName = split[2];
+
+                    String time = split[3];
+
+                    Task event = new Event(taskName, time);
+                    taskList.add(event);
+
+                    isDone = Boolean.parseBoolean(split[1]);
+                    if (isDone) {
+                        event.markAsDone();
+                    }
+                    break;
+                default:
+                    break;
+                }
+            }
+        } catch (FileNotFoundException e) {
```

Comment:
This method is particularly long. Perhaps you could extract out to another method?